### PR TITLE
server:  remove the scheduler completely that does not enable by default

### PR DESCRIPTION
**Pull request by nolouch**
There is too many schedulers print When using the `pdctl` show the config. this pr fix it.

### CODE REVIEW

File: server/option.go [hunk 1]
```diff
@@ -177,8 +177,12 @@ func (o *scheduleOption) RemoveSchedulerCfg(name string) error {
 			return errors.Trace(err)
 		}
 		if tmp.GetName() == name {
-			schedulerCfg.Disable = true
-			v.Schedulers[i] = schedulerCfg
+			if IsDefaultScheduler(tmp.GetType()) {
+				schedulerCfg.Disable = true
+				v.Schedulers[i] = schedulerCfg
+			} else {
+				v.Schedulers = append(v.Schedulers[:i], v.Schedulers[i+1:]...)
```
Comment (Connor1996) [inline server/option.go]: it seems that it will skip the origin `i+1` item after appending

Comment (nolouch) [reply server/option.go]: it should be [...)

**Comment by disksing** [comment]
LGTM.

File: server/option.go (hunk 1 above)
Comment (Connor1996) [reply server/option.go]: what I mean is in next loop `i` will be `i+1`, but after appending the now `i+1` item is actually the origin `i+2` item, namely it skip origin `i+1` item

Comment (nolouch) [reply server/option.go]: it will return directly later, no next loop.

Comment (Connor1996) [reply server/option.go]: Oh~ I got it.

**Pull request merged**